src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -512,4 +518,20 @@ protected T amend(String optionName, Object value) {
         setCapability(optionName, value);
         return (T) this;
     }
+
+    @Override
+    public Map<String, Object> asMap() {
+        Map<String, Object> toReturn = new TreeMap<>();
+
+        Map<String, Object> capsMap = super.asMap();
+        capsMap.forEach((key, value) -> {
```

Comment:
I think we can build toReturn without having the intermediate `capsMap` variable, e.g.

```
Map<String, Object> toReturn = super.asMap().entrySet().stream().map ....
```

Reply:
I pretty much just copied it from Selenium code, which I think is what happens when you use Selenium 4 with current Appium code? And Selenium browser options I think did more things with the intermediate step, so maybe it can be replaced with your suggestion here, I can try.

Reply:
So, I can't figure out how to get the stream to work. This is what I'm working with right now, which removes the redundant `capsMap`, but still uses `toReturn` and `forEach` directly:

```java
        super.asMap().forEach((key, value) -> {
            AcceptedW3CCapabilityKeys acceptedW3CCapabilityKeys = new AcceptedW3CCapabilityKeys();
            if (acceptedW3CCapabilityKeys.test(key) || key.startsWith(APPIUM_PREFIX)) {
                toReturn.put(key, value);
            } else {
                toReturn.put(APPIUM_PREFIX + key, value);
            }
        });
```

Reply:
Ok, got a stream figured out. Let me know what you think.